### PR TITLE
feat(VBC-326): common interfaces for wishlists

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
Common interfaces for useGuestWishlist and useWishlist

## Description

There were some differences in implementation of useGuestWishlist and useWishlist actions. This PR makes them the same.

## Related Issue

VBC-326

## Motivation and Context

To be able to use `isAuthenticated ? useGuestWishlist() : useWishlist();`

## How Has This Been Tested?

Tested locally

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/components/WishlistSidebar.vue [hunk 1]
```diff
@@ -105,7 +105,7 @@ export default defineComponent({
     const wishlistHelpers = useWishlistData();
     const { isAuthenticated } = useUser();
     const { isWishlistSidebarOpen, toggleWishlistSidebar } = useUiState();
-    const { wishlist, removeItem, load: loadWishlist } = useGuestWishlist('guest-wishlist');
+    const { wishlist, removeItem, load: loadWishlist } = isAuthenticated ? useGuestWishlist() : useWishlist();
```
Comment (dhargitai) [inline packages/theme/components/WishlistSidebar.vue:108]: I guess a `!` is missing before the `isAuthenticated`, right?

File: packages/theme/pages/Category.vue [hunk 2]
```diff
@@ -320,7 +340,7 @@ export default defineComponent({
       addItem: addItemToWishlist,
       isInWishlist,
       removeItem: removeItemFromWishlist
-    } = useGuestWishlist('guest-wishlist');
+    } = isAuthenticated ? useGuestWishlist() : useWishlist();
```
Comment (dhargitai) [inline packages/theme/pages/Category.vue:343]: Same as above, I think you want to use the guest wishlist on the unauthenticated state.

File: packages/theme/pages/Category.vue [hunk 3]
```diff
@@ -171,14 +172,30 @@
...
+                  currentWishlist: wishlist,
+                  product
                 })
               "
               :qty="1"
               @input="productsQuantity[product.id] = $event"
               class="products__product-card-horizontal"
-              @click:wishlist="null"
+              @click:wishlist="
+                InWishlist({
```
Comment (dhargitai) [inline packages/theme/pages/Category.vue:183]: I think this would be `isInWishlist`

**Review by dhargitai — CHANGES_REQUESTED** [review]
Looks good, I only had some small comments.

File: packages/theme/pages/Category.vue (hunk 3 above)
Comment (WojtekTheWebDev) [reply packages/theme/pages/Category.vue:183]: Right, thanks!

File: packages/theme/pages/Category.vue (hunk 2 above)
Comment (WojtekTheWebDev) [reply packages/theme/pages/Category.vue:343]: Indeed, thanks :)

File: packages/theme/components/WishlistSidebar.vue (hunk 1 above)
Comment (WojtekTheWebDev) [reply packages/theme/components/WishlistSidebar.vue:108]: Right :)

**Review by dhargitai — APPROVED** [review]
👍

**Pull request merged**